src/GraphQL/Resolver/DomainContentResolver.php

```diff
@@ -136,24 +136,19 @@ public function resolveDomainFieldValue(Content $content, $fieldDefinitionIdenti
 
     public function resolveDomainRelationFieldValue(Field $field, $multiple = false)
     {
-        if (!$field->value instanceof FieldType\RelationList\Value) {
-            throw new UserError("$field->fieldTypeIdentifier is not a RelationList field value");
-        }
-
-        if ($multiple) {
-            if (count($field->value->destinationContentIds) > 0) {
-                return $this->contentLoader->find(new Query(
-                    ['filter' => new Query\Criterion\ContentId($field->value->destinationContentIds)]
-                ));
-            } else {
-                return [];
-            }
+        if ($field->value instanceof FieldType\RelationList\Value) {
+            $destinationContentIds = $field->value->destinationContentIds;
+        } else if ($field->value instanceof FieldType\Relation\Value) {
+            $destinationContentIds = [$field->value->destinationContentId];
         } else {
-            return
-                isset($fieldValue->destinationContentIds[0])
-                    ? $this->contentLoader->findSingle(new Query\Criterion\ContentId($field->value->destinationContentIds[0]))
-                    : null;
+            throw new UserError('\$field does not contain a RelationList or Relation Field value');
         }
+
+        $contentItems = $this->contentLoader->find(new Query(
+            ['filter' => new Query\Criterion\ContentId($destinationContentIds)]
```

Comment:
Tested this with Solr enabled and it will throw an exception when $destinationContentIds is an empty array, a suggestion would be to add the following before "$contentItems = ..."  `if(empty($destinationContentIds)) {
            return $multiple ? [] : null;
        }`

Reply:
Thank you, very good remark @crosa7. It should be fixed now.